packages/template-retail-react-app/app/utils/sf-payments-utils.js

```
@@ -78,8 +78,8 @@ export const transformShippingMethods = (
         id: method.id,
         name: method.name,
         classOfService: method.description,
-        shippingFee: typeof method.price === 'number' ? method.price.toString() : method.price,
-        currencyIsoCode: basket.currency
+        // amount is the price of the shipping method (or shipping fee)
+        amount: typeof method.price === 'number' ? method.price.toString() : method.price
```

do we need this type check?

---

It was always there.  I just changed shippingFee to amount. If the API always returns a number, this is sufficient. If the string test case was added defensively and isn't needed, we can remove.

shippingFee: typeof method.price === 'number' ? method.price.toString() : method.price,

---

If we're concerned the API will return invalid data then let's validate it. You could validate it here, throw if invalid, and have the caller handle that case by calling back to the SDK with fail status. IMO this isn't a case we need to validate, so I'd just call `toString`. You could use `method.price?.toString()` if you want to at least  avoid calling the function on `null`/`undefined` but even that may be unnecessary.